### DISCUSSION TITLE
How to import CSV files to populate an existing Database

### DISCUSSION THREAD

**Discussion by bridei** [Budibase/budibase#3790] [Ideas]
Hello
In a Budibase app self hosting, i need to allow the users to import rows in an exisiting database, using a CSV import dialog. The CSV to import will have the same headers as the table to import
Is there any way to do it with budibase?

Thanks so much!

**Comment by mjashanks** [comment]
Hi There.

Are talking about the end-users of your app importing data? If so, it's not possible at the moment.

You can import data via the Budibase builder of course, just not through the apps themselves.

**Reply by bridei** [reply]
Thanks
The question was throught the app themselves...
And, it is possible for the end users to populate a form importing a CSV? ( more than 20 fields to enter manually is difficult for them)?

**Reply by mjashanks** [reply]
Unfortunately not. I like the idea though, makes sense. Converting this to an idea.

**Reply by gvlx** [reply]
Could this be done through an automation?

Creating a 'loop' or 'repeater' block which receives a list and applies each element to the blocks afterwards?

**Reply by mjashanks** [reply]
We really need a loop automation block - it would be so useful. Even with that, we would also need the automation webhook trigger to accept different content types (i.e. CSV).

But I agree with you, this would be a great way to import CSV, as it would give use the transformation abilities of automations when importing. Hadn't thought of doing it this way before - thanks!

**Reply by mmalessa80** [reply]
Looking for this type of solution as well. I would be happy with a file input field that does not need to be tied to a form that would allow me to parse the contents of a CSV file and then create a JSON array from it that I can loop over to create records.

**Reply by jgngo** [reply]
This is great idea!

**Reply by kasgithub8** [reply]
@jgngo there is a plugin for this now!
https://github.com/mjashanks/budibase-csv-import
